library/ziti.c

```diff
@@ -160,6 +184,222 @@ int load_tls(nf_config *cfg, tls_context **ctx) {
     return ZITI_OK;
 }
 
+int NF_enroll(const char* jwt_file, uv_loop_t* loop, nf_enroll_cb external_enroll_cb) {
+    init_debug();
+
+    int ret;
+    struct enroll_cfg_s *ecfg = NULL;
+    uv_timeval64_t start_time;
+    uv_gettimeofday(&start_time);
+
+    struct tm *start_tm = gmtime((const time_t)&start_time.tv_sec);
+    char time_str[32];
+    strftime(time_str, sizeof(time_str), "%FT%T", start_tm);
+
+    ZITI_LOG(INFO, "ZitiSDK version %s @%s(%s) starting at (%s.%03d)",
+            ziti_get_version(false), ziti_git_commit(), ziti_git_branch(),
+            time_str, start_time.tv_usec/1000);
+
+    PREP(ziti);
+    
+    nf_config *cfg = NULL;
+    tls_context *tls = NULL;
+
+    ecfg = calloc(1, sizeof(enroll_cfg));
+    ecfg->external_enroll_cb = external_enroll_cb;
+
+    TRY(ziti, load_jwt(jwt_file, ecfg, &ecfg->zejh, &ecfg->zej));
+    if (DEBUG <= ziti_debug_level) {
+        dump_ziti_enrollment_jwt_header(ecfg->zejh, 0);
+        dump_ziti_enrollment_jwt(ecfg->zej, 0);
+    }
+
+    // JWT validation start
```

Comment:
This function is huge.

Reply:
![image](https://user-images.githubusercontent.com/919888/77163322-3673af80-6a84-11ea-969a-2bb3652b3fe3.png)